src/test/fuzz/fuzz.h

```diff
@@ -9,6 +9,6 @@
 #include <vector>
 
 
-void test_one_input(std::vector<uint8_t> buffer);
+void test_one_input(const std::vector<uint8_t>& buffer);
```

Comment:
Looks fine to do this change, even if we were to switch to a different class (like https://cs.chromium.org/chromium/src/third_party/libFuzzer/src/utils/FuzzedDataProvider.h?rcl=b9f51dc8c98065df0c8da13c051046f5bab833db), the input data remains read-only.

Reply:
Yes, fuzzing inputs are passed read-only by convention for all in-memory fuzzers I've ever used. And `FuzzedDataProvider` is great :)

Reply:
Could split up this commit, so that other pull request don't conflict with this one?

Reply:
@practicalswift ^ ;)

Reply:
I've now moved this change to a separate PR to make this PR non-conflicting with others. I hope that was the correct reading of the suggestion :)